File: packages/radar/frontend/src/types/configuration.ts
/**
 * Configuration types - aligned with backend schemas (configuration.py)
 */

export type ConfigCategory = "thematic" | "scoring" | "delivery" | "llm" | "general" | "sources";

export interface Configuration {
  id: string;
  category: string;
  key: string;
  value: Record<string, unknown>;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export type ConfigValue = Record<string, unknown>;

export interface ConfigurationUpdatePayload {
  value: ConfigValue;
}

File: packages/radar/backend/app/models/configuration.py
from typing import Any

from sqlalchemy import CheckConstraint, String, Text, UniqueConstraint
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Configuration(Base):
    """SQLAlchemy model for the configuration table."""

    __tablename__ = "configuration"

    category: Mapped[str] = mapped_column(String(20), nullable=False)
    key: Mapped[str] = mapped_column(String(100), nullable=False)
    value: Mapped[dict[str, Any]] = mapped_column(JSONB, nullable=False)
    description: Mapped[str | None] = mapped_column(Text)

    __table_args__ = (
        UniqueConstraint("category", "key", name="uq_configuration_category_key"),
        CheckConstraint(
            "category IN ('thematic', 'scoring', 'delivery', 'llm', 'general', 'sources')",
            name="chk_config_category",
        ),
    )

    def __repr__(self) -> str:
        return f"<Configuration(id={self.id}, category='{self.category}', key='{self.key}')>"
